.bats/entities.sh

```diff
@@ -8,33 +8,61 @@ cd $HOME/ardrive-cli/
 cache="$(yarn ardrive list-all-drives -w $WALLET)"
 
 # RAW, one line, no brackets and no quotes
-pubDriveList="$(echo $cache | jq -rc '.[] | select(.drivePrivacy == "public") | .driveId')"
+allDriveIDs="$(echo $cache | jq -rc '.[] | select(.drivePrivacy == "public") | .driveId')"
 
 #Iterates over list of driveIDs from wallet.
-for publicDrive in ${pubDriveList[@]}; do
-    aux="$(yarn ardrive list-drive -d $publicDrive)"
-    numberFiles="$(echo $aux | jq -r '[.[] | select(.entityType == "file")] | length')"
+found=0
+for publicDriveId in ${allDriveIDs[@]}; do
+    driveContent="$(yarn ardrive list-drive -d $publicDriveId)"
     if [[ $? -ne 0 ]]; then
         printf "Error loading entities. Please check your wallet! \n"
         exit 1
     fi
-    if [ "$numberFiles" -ge "2" ]; then
-        fileID="$(echo $aux | jq -r '[.[] | select(.entityType == "file")][0] | .entityId')"
-        fileSize="$(echo $aux | jq -r '[.[] | select(.entityType == "file")][0] | .size')"
-        fileName="$(echo $aux | jq -r '[.[] | select(.entityType == "file")][0] | .name')"
-        fileNameAlt="$(echo $aux | jq -r '[.[] | select(.entityType == "file")][1] | .name')"
-        folderID="$(echo $aux | jq -r '[.[] | select(.entityType == "file")][0] | .parentFolderId')"
-        rootID="$(echo $aux | jq -r '[.[] | select(.parentFolderId == "root folder")][0] | .entityId')"
-        driveID="$(echo $aux | jq -r '[.[] | select(.entityType == "file")][0] | .driveId')"
+    allFolderIDs="$(echo $driveContent | jq -rc '.[] | select(.entityType == "folder") | .entityId')"
+
+    for someFolderId in ${allFolderIDs[@]}; do
```

Comment:
All of this is getting messy and complicated. We really need to get ArLocal up and running soon to create data sets on the fly.

Reply:
I think my chief complaint is that there's not a simple explanation in a comment up above about what we're trying to accomplish here. The script description mentions `sets ENV variables using the values from the first valid Public file` which is not only partially valid. And also there's a whole workflow here dedicated to figuring out a structure, but that's not explicitly spelled out above. So I end up needing to read a bunch of code and conditionals to understand what you're trying to find.

Reply:
Added comments for clarification: f5cac008545cb38f1cb87a91b2859e579bba64df